src/mpid/ch4/shm/posix/posix_coll_nb_release_gather.h

```diff
@@ -0,0 +1,149 @@
+/*
+ *
+ *  Copyright (C) by Argonne National Laboratory.
+ *      See COPYRIGHT in top-level directory.
+ */
+
+#ifndef POSIX_COLL_NB_RELEASE_GATHER_H_INCLUDED
+#define POSIX_COLL_NB_RELEASE_GATHER_H_INCLUDED
+
+#include "mpiimpl.h"
+#include "algo_common.h"
+#include "nb_release_gather.h"
+
+/* Intra-node Ibcast is implemented as a release step followed by gather step in release_gather
+ * framework. The actual data movement happens in release step. Gather step makes sure that
+ * the shared bcast buffer can be reused for next bcast call. Release gather framework has
+ * multitple cells in bcast buffer, so that the copying in next cell can be overlapped with
+ * copying out of previous cells (pipelining).
+ */
+MPL_STATIC_INLINE_PREFIX int MPIDI_POSIX_ibcast_release_gather(void *buffer, MPI_Aint count,
+                                                               MPI_Datatype datatype, int root,
+                                                               MPIR_Comm * comm_ptr,
+                                                               MPIR_TSP_sched_t * sched)
+{
+    MPIR_FUNC_ENTER;
+    int i, my_rank, num_chunks, chunk_count_floor, chunk_count_ceil;
+    MPI_Aint nbytes;
+    MPI_Aint true_lb = 0, true_extent, type_size;
+    void *ori_buffer = buffer;
+    MPI_Datatype ori_datatype = datatype;
+    int offset = 0, is_contig, ori_count = count;
+    int mpi_errno = MPI_SUCCESS;
+    MPIR_Request *req = NULL;
+    int *sink_id, vtx_id;
+
+    MPIR_CHKLMEM_DECL(1);
+
+    /* If there is only one process or no data, return */
+
+    if (count == 0 || (MPIR_Comm_size(comm_ptr) == 1)) {
+        goto fn_exit;
+    }
```

Comment:
We should handle these trivial cases in the C binding layer. Intracomm size of 1 is already done. Just need to add a patch for 0 count.

Reply:
https://github.com/pmodels/mpich/pull/5666 will add this trivial case in the binding layer.

Reply:
> The large commit has a lot going on. I'd really like to see an effort to split it up into individual pieces in order to review. See my suggestion inline.

OK. I will work on breaking down the large commit. But this PR has a dependency on another PR that @zhenggb72 is working on. So, this one needs to wait for a while.